lib/ssl/src/ssl.erl

```diff
@@ -799,23 +809,263 @@ certificate chain validating the CRLs.
 
 The CRLs will be fetched from a local or external cache. See
 `m:ssl_crl_cache_api`.
-""".
-
--type policy_opt() :: {policy_set, [public_key:oid()]} | {explicit_policy, boolean()} |  {inhibit_policy_mapping, boolean()} | {inhibit_any_policy, boolean()}.
--doc """
-Configure X509 certificate policy handling for the certificate path validation process
-see [(public_key:pkix_path_validation/3) ](`public_key:pkix_path_validation/3`) for
-further explanation.
 
 """.
+-type common_option_cert() :: {certs_keys, CertsKeys::[cert_key_conf()]} |
+                              {depth, AllowedCertChainLen::pos_integer()} |
+                              {verify_fun, Verify::{Verifyfun :: fun(), InitialUserState :: any()}} |
+                              {cert_policy_opts, PolicyOpts::[{policy_set, [public_key:oid()]} |
+                                                              {explicit_policy, boolean()} |
+                                                              {inhibit_policy_mapping, boolean()} |
+                                                              {inhibit_any_policy, boolean()}]} |
+                              {crl_check, Check::boolean() | peer | best_effort} |
+                              {crl_cache, crl_cache_opts()} |
+                              {partial_chain, anchor_fun()}.
+
 
 -doc(#{title =>
-           <<"TLS/DTLS OPTION DESCRIPTIONS - COMMON for SERVER and CLIENT">>}).
--type crl_check()                :: boolean() | peer | best_effort.
+           <<"Common Options">>}).
 -doc """
-Specify how to perform lookup and caching of certificate revocation lists.
-`Module` defaults to `m:ssl_crl_cache` with `DbHandle `being `internal` and an
-empty argument list.
+Options common to both client and server side pre TLS-1.3.
+
+- **\{eccs, NamedCurves}** -
+Elliptic curves that can be use in pre TLS-1.3 key exchange.
+
+- **\{secure_renegotiate, SecureRenegotiate}** -
+Specifies if to reject renegotiation attempt that does not live up to
+[RFC 5746](http://www.ietf.org/rfc/rfc5746.txt). By default `secure_renegotiate`
+is set to `true`, that is, secure renegotiation is enforced. If set to `false`
+secure renegotiation will still be used if possible, but it falls back to
+insecure renegotiation if the peer does not support
+[RFC 5746](http://www.ietf.org/rfc/rfc5746.txt).
+
+- **\{user_lookup_fun, {LookupFun, UserState}}** -
+The lookup fun is to defined as follows:
+
+```erlang
+fun(psk, PSKIdentity :: binary(), UserState :: term()) ->
+	{ok, SharedSecret :: binary()} | error;
```

Comment:
```suggestion
    {ok, SharedSecret :: binary()} | error;
```

Spaces are preferred to tabs in examples. You don't have to fix these, but can if you want to. There are many places where this is done, I've only commented on this one.